algorithms/linfa-clustering/src/dbscan/hyperparameters.rs

```diff
@@ -7,54 +8,70 @@ use serde_crate::{Deserialize, Serialize};
     derive(Serialize, Deserialize),
     serde(crate = "serde_crate")
 )]
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Debug)]
 /// The set of hyperparameters that can be specified for the execution of
 /// the [DBSCAN algorithm](struct.Dbscan.html).
-pub struct DbscanHyperParams<F: Float> {
+pub struct DbscanHyperParams<F: Float, D: Distance<F>, N: NearestNeighbour> {
     /// Distance between points for them to be considered neighbours.
     tolerance: F,
     /// Minimum number of neighboring points a point needs to have to be a core
     /// point and not a noise point.
     min_points: usize,
+    /// Distance metric used in the DBSCAN calculation
+    dist_fn: D,
+    /// Nearest neighbour algorithm used for range queries
+    nn_impl: N,
 }
 
 /// Helper struct used to construct a set of hyperparameters for [DBSCAN algorithm](struct.Dbscan.html).
-pub struct DbscanHyperParamsBuilder<F: Float> {
-    tolerance: F,
-    min_points: usize,
+pub struct DbscanHyperParamsBuilder<F: Float, D: Distance<F>, N: NearestNeighbour> {
+    pub(crate) tolerance: F,
+    pub(crate) min_points: usize,
+    pub(crate) dist_fn: D,
+    pub(crate) nn_impl: N,
 }
 
-impl<F: Float> DbscanHyperParamsBuilder<F> {
+impl<F: Float, D: Distance<F>, N: NearestNeighbour> DbscanHyperParamsBuilder<F, D, N> {
     /// Distance between points for them to be considered neighbours.
     pub fn tolerance(mut self, tolerance: F) -> Self {
         self.tolerance = tolerance;
         self
     }
 
+    /// Nearest neighbour algorithm used for range queries
+    pub fn nn_impl(mut self, nn_impl: N) -> Self {
```

Comment:
I'd not use the `_impl` here because it's a reserved word and may confuse reader, perhaps `nn_backend` or `nn_algo` or just plain `nn`?

Reply:
I'm going to use `nn_algo` instead.